## tl;dr

PN17 placed one arbitrary anchor at `400,000,000,000`, decompressed the complete lower-child phase field once, and
sealed `+19` as the first quiet factor ridge before checking a target prime label. Independent validation confirmed
that `400,000,000,019` is prime and that every intervening integer is composite (`26/26` checks).

The full child vector works exactly because it is the standard segmented-sieve collision mask in ARA coordinates.
Three simple scalar A/B averages did not select +19. The remaining new-theory problem is therefore compression of
the child web without erasing its periods and phases.


In [1]:
from pathlib import Path
import hashlib, json

HERE = Path(r'F:\SystemFormulaFolder\GIT\ARA-GIT\analysis\primes')
prediction = json.loads((HERE / 'PN17_ONE_SHOT_LOCAL_RIDGE_PREDICTION.json').read_text(encoding='utf-8'))
validation = json.loads((HERE / 'PN17_ONE_SHOT_LOCAL_RIDGE_VALIDATION.json').read_text(encoding='utf-8'))
diagnostic = json.loads((HERE / 'PN17_SCALAR_RIDGE_DIAGNOSTIC.json').read_text(encoding='utf-8'))
print('Sealed candidate:', prediction['target']['predicted_integer'])
print('Correction:', prediction['target']['correction'])
print('Independent first-prime validation:', validation['candidate_is_first_prime_above_anchor'])
print('Checks:', validation['passed_count'], '/', validation['check_count'])


Sealed candidate: 400000000019
Correction: 19
Independent first-prime validation: True
Checks: 26 / 26


## Context & Methods

### Key Assumptions

- “Prime ridge” is operationalized as a quiet factor ridge: no lower prime child collides through `sqrt(n)`.
- The complete child vector is permitted; no scalar TE-ARA aggregation law is assumed.
- The target builder may generate lower prime children but may not call a target primality function or read nearby
  target prime labels.
- The result is compared with a standard segmented sieve, which is mathematically the same collision field.

For each offset `t`, the calculation counts lower-child collisions. The correction is the first positive offset with
count zero.


In [2]:
print('Development anchors')
print('anchor | correction | prediction | lower children | pass')
for row in prediction['development']:
    print(f"{row['anchor']:>12,} | {row['correction']:>10} | {row['predicted_integer']:>12,} | "
          f"{row['child_count']:>14,} | {row['matches_development_control']}")


Development anchors
anchor | correction | prediction | lower children | pass
 100,000,000 |          7 |  100,000,007 |          1,229 | True
1,000,000,000 |          7 | 1,000,000,007 |          3,401 | True
10,000,000,000 |         19 | 10,000,000,019 |          9,592 | True
100,000,000,000 |          3 | 100,000,000,003 |         27,293 | True


## Data

The source is exact integer arithmetic. The target block contains 65,536 offsets and 51,526 lower prime children,
ending at gate 632,447. The binary collision field was sealed with the prediction and reconstructed independently.
The equal-gap control uses the already-opened PN7C R11 actual-prime gap packet.


In [3]:
target = prediction['target']
print('Target anchor:', f"{target['anchor']:,}")
print('Child count:', f"{target['child_count']:,}")
print('Child ceiling:', f"{target['child_ceiling']:,}")
print('First quiet offset:', target['correction'])
print('Sealed integer:', f"{target['predicted_integer']:,}")
print('Quiet offsets in full block:', f"{target['quiet_offsets_in_block']:,}")
print('Collision-field SHA-256:', target['collision_field_sha256'])


Target anchor: 400,000,000,000
Child count: 51,526
Child ceiling: 632,447
First quiet offset: 19
Sealed integer: 400,000,000,019
Quiet offsets in full block: 2,463
Collision-field SHA-256: 839A8F2143C7652FBBDF18013F08E60E334A323623A4289708A692577E7ADEFD


## Results

The target passed both independent deterministic Miller-Rabin and full trial division, and no earlier integer was
prime. Baselines are reported honestly: odd scanning would inspect 10 odd candidates; a p29 wheel would retain five;
the complete ARA calculation used 51,526 child phases and is identical to a local segmented sieve.


In [4]:
print('Baselines:', prediction['baselines'])
print('Equal-gap control:', prediction['equal_gap_falsification_control'])
print('\nSimple scalar ridge diagnostics')
for name, record in diagnostic['methods'].items():
    print(name, {
        'best_offset': record['best_scalar_ridge_offset_in_opened_range'],
        'best_is_quiet': record['best_scalar_ridge_is_quiet'],
        'prime_rank': record['sealed_prime_rank_by_scalar_ridge_error'],
        'anchor_error': record['anchor_ridge_error'],
    })


Baselines: {'odd_scan_candidates_through_prediction': 10, 'p29_wheel_candidates_through_prediction': 5, 'full_child_phases_used': 51526, 'standard_segmented_sieve_equivalence': True}
Equal-gap control: {'source': 'PN7C_R11_TARGET_GAPS.npz', 'source_sha256': 'D60EEDFA2F3A5DF4C8FA45B45D2B478EDB39B6D54001302F84041989C8D0CF2F', 'events': 39475589, 'exact_equal_gap_hit_rate': 0.02093696436043044, 'mean_absolute_gap_error': 23.23636402233289, 'median_absolute_gap_error': 16.0, 'within_2_gap_units': 0.1148014536274557, 'within_4_gap_units': 0.19740787148229758, 'mean_actual_outgoing_gap': 25.332110434121706}

Simple scalar ridge diagnostics
equal_child {'best_offset': 0, 'best_is_quiet': False, 'prime_rank': 10, 'anchor_error': 0.0013127655928243387}
log_child {'best_offset': 14, 'best_is_quiet': False, 'prime_rank': 16, 'anchor_error': 0.0009982711349195705}
inverse_period {'best_offset': 7, 'best_is_quiet': False, 'prime_rank': 6, 'anchor_error': 0.21753445634702862}


## Takeaways

1. Dylan's intended local direction is executable: begin at the desired scale, decompress the child web once, and
   solve for the nearest quiet ridge.
2. The result is exact and cleanly sealed, but its current implementation is a segmented sieve crosswalk rather than
   a faster prime algorithm.
3. A prime node is not generally raw incoming-gap equals outgoing-gap; that control hits only about 2.09% on R11.
4. Averaging the child A/B states erases the modular locations. The successful step is recursively retaining the
   identities inside “Other.”
5. The next test needs a predeclared low-dimensional coupling law that predicts the correction without rebuilding all
   51,526 collision lanes.
